src/demo-app/tree/dynamic-tree-demo/dynamic-database.ts

```diff
@@ -100,13 +100,22 @@ export class DynamicDataSource {
     }
     node.isLoading = true;
 
+    let count = 0;
+
     setTimeout(() => {
       if (expand) {
         const nodes = children.map(name =>
           new DynamicFlatNode(name, node.level + 1, this.database.isExpandable(name)));
         this.data.splice(index + 1, 0, ...nodes);
       } else {
-        this.data.splice(index + 1, children.length);
+        for (let i = index + 1; i < this.data.length; i++) {
```

Comment:
Is this better? 
```ts
let count = 0;
for (let i = index + 1; i < this.data.length && this.data[i].level > node.level; i++, count++) {}?
this.data.splice(index + 1, count);
```

Reply:
Yes, thanks! This definitely saves space